src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexFindOptimizations.cs

```diff
@@ -179,6 +182,12 @@ public RegexFindOptimizations(RegexNode root, RegexOptions options)
                         FindMode = (fixedDistanceSets.Count == 1 && fixedDistanceSets[0].Distance == 0) ? FindNextStartingPositionMode.LeadingSet_LeftToRight
                             : FindNextStartingPositionMode.FixedDistanceSets_LeftToRight;
                         _asciiLookups = new uint[fixedDistanceSets.Count][];
+                        if (FindMode == FindNextStartingPositionMode.FixedDistanceSets_LeftToRight || fixedDistanceSets[0].Chars is null)
+                        {
+                            // FixedDistanceSets and LeadingSet with null Chars are not vectorized.
```

Comment:
I don't understand this condition.  It's jumping to PreferLeadingMultiStringIfAvailable if FindMode == FindNextStartingPositionMode.FixedDistanceSets_LeftToRight, regardless of whether fixedDistanceSets[0].Chars is null, but FixedDistanceSets is absolutely vectorizable when Chars is not null.

Reply:
Answered in https://github.com/dotnet/runtime/pull/71588#discussion_r916656081.

Reply:
I removed the first part of the `||`. We now only check if `Chars` is null.